### PR TITLE
Creating kube manifest dirs on node provision.

### LINKED ISSUE #3506: create /etc/kubernetes/manifests on worker nodes
Lots of spurious kubelet messages:

```
Jul 18 18:13:05 k8s-agentmd-34767551-2 kubelet[8431]: E0718 18:13:05.032451    8431 file_linux.go:61] Unable to read config path "/etc/kubernetes/manifests": path does not exist, ignoring```

### PR DESCRIPTION
**Pull request by tariq1890**
**What this PR does / why we need it**: Fixes #3506 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
@CecileRobertMichon your thoughts on if there's somewhere more elegant to put this? It seems like this directory is being created on master implicitly by cloud-init because it has files that are in there (i.e., cloud-init rationalizes a filepath that doesn't yet exist by creating directories if needed)

**Comment by CecileRobertMichon** [comment]
I think we should put this in custom data (cloud-init), it could be part of the runcmd (https://github.com/Azure/acs-engine/blob/master/parts/k8s/kubernetesagentcustomdata.yml#L219) or in "write_files" if it's possible to create a directory

**Comment by jackfrancis** [comment]
I read through this and it suggests we write_files isn't able to create an empty dir:

https://cloudinit.readthedocs.io/en/latest/topics/examples.html#

**Comment by jackfrancis** [comment]
`runcmd` is too non-deterministic as far as I'm concerned

File: parts/k8s/kubernetescustomscript.sh [hunk 1]
```diff
@@ -67,6 +67,7 @@ installNetworkPlugin
 installContainerd
 extractHyperkube
 ensureRPC
+createKubeManifestDir
```
Comment (jackfrancis) [inline parts/k8s/kubernetescustomscript.sh]: Note: if we *do* decide to keep this in CSE, let's move this fun invocation into the `if $FULL_INSTALL_REQUIRED; then` block. We should create this directory during VHD image creation process.

Comment (CecileRobertMichon) [reply parts/k8s/kubernetescustomscript.sh]: `if $FULL_INSTALL_REQUIRED;` is for deployment not using the AKS image so I don't think we should move it, we want this to run on every deployment

**Comment by CecileRobertMichon** [comment]
Keeping it in CSE is probably fine then

**Comment by CecileRobertMichon** [comment]
If we want to do this during VHD creation we'll need to call the function from https://github.com/Azure/acs-engine/blob/master/packer/install-dependencies.sh

**Comment by CecileRobertMichon** [comment]
@jackfrancis thinking about this more, I don't think we need to do this in VHD since mkdir is a primitive operation and we don't expect any delay/flakiness from it. I'm open to being convinced of the opposite though.

**Pull request merged**